pkg/volume/portworx/portworx_util.go

```diff
@@ -56,11 +56,14 @@ func (util *PortworxVolumeUtil) CreateVolume(p *portworxVolumeProvisioner) (stri
 	requestGB := int(volume.RoundUpSize(capacity.Value(), 1024*1024*1024))
 
 	specHandler := osdspec.NewSpecHandler()
-	spec, err := specHandler.SpecFromOpts(p.options.Parameters)
-	if err != nil {
-		glog.Errorf("Error parsing parameters for PVC: %v. Err: %v", p.options.PVC.Name, err)
-		return "", 0, nil, err
+	spec, _ := specHandler.SpecFromOpts(p.options.Parameters)
+	if spec == nil {
+		spec = specHandler.DefaultSpec()
```

Comment:
> Now you pass the default spec with HaLevel = 1, Cos = 0 and with repl and io_priority labels to driver.Create. Is this correct? 
Yes. 
> Will driver.Create parse the labels again and override spec.HaLevel and spec.Cos?

Yes, our latest server side implementation of driver.Create will overide the spec with entries given in the labels. 

> So what's the point of calling SpecFromOpts if everything can be passed as labels?

If everyone is running the latest Portworx version, we don't need to call `SpecFromOpts`
This is done to handle the case, someone updates to the latest k8s but doesn't update Portworx. Since previous versions of Portworx do not overide spec from labels, I am still calling `SpecFromOpts` for backward compatibility. 

The err is ignored because going forward, we don't want any StorageClass parsing errors in the k8s driver to block volume creation.